packages/i18n/src/create-i18n.js

```diff
@@ -0,0 +1,195 @@
+	 *
+	 * @param {string} text     Text to translate.
+	 * @param {string} [domain] Domain to retrieve the translated text.
+	 *
+	 * @return {string} Translated text.
+	 */
+	const __ = ( text, domain ) => {
+		return dcnpgettext( domain, undefined, text );
+	};
+
+	/**
+	 * Retrieve translated string with gettext context.
+	 *
+	 * @see https://developer.wordpress.org/reference/functions/_x/
+	 *
+	 * @param {string} text     Text to translate.
+	 * @param {string} context  Context information for the translators.
+	 * @param {string} [domain] Domain to retrieve the translated text.
+	 *
+	 * @return {string} Translated context string without pipe.
+	 */
+	const _x = ( text, context, domain ) => {
+		return dcnpgettext( domain, context, text );
+	};
+
+	/**
+	 * Translates and retrieves the singular or plural form based on the supplied
+	 * number.
+	 *
+	 * @see https://developer.wordpress.org/reference/functions/_n/
+	 *
+	 * @param {string} single   The text to be used if the number is singular.
+	 * @param {string} plural   The text to be used if the number is plural.
+	 * @param {number} number   The number to compare against to use either the
+	 *                          singular or plural form.
+	 * @param {string} [domain] Domain to retrieve the translated text.
+	 *
+	 * @return {string} The translated singular or plural form.
+	 */
+	const _n = ( single, plural, number, domain ) => {
+		return dcnpgettext( domain, undefined, single, plural, number );
+	};
+
+	/**
+	 * Translates and retrieves the singular or plural form based on the supplied
+	 * number, with gettext context.
+	 *
+	 * @see https://developer.wordpress.org/reference/functions/_nx/
+	 *
+	 * @param {string} single   The text to be used if the number is singular.
+	 * @param {string} plural   The text to be used if the number is plural.
+	 * @param {number} number   The number to compare against to use either the
+	 *                          singular or plural form.
+	 * @param {string} context  Context information for the translators.
+	 * @param {string} [domain] Domain to retrieve the translated text.
+	 *
+	 * @return {string} The translated singular or plural form.
+	 */
+	const _nx = ( single, plural, number, context, domain ) => {
+		return dcnpgettext( domain, context, single, plural, number );
+	};
+
+	/**
+	 * Check if current locale is RTL.
+	 *
+	 * **RTL (Right To Left)** is a locale property indicating that text is written from right to left.
+	 * For example, the `he` locale (for Hebrew) specifies right-to-left. Arabic (ar) is another common
+	 * language written RTL. The opposite of RTL, LTR (Left To Right) is used in other languages,
+	 * including English (`en`, `en-US`, `en-GB`, etc.), Spanish (`es`), and French (`fr`).
+	 *
+	 * @return {boolean} Whether locale is RTL.
+	 */
+	const isRTL = () => {
+		return 'rtl' === _x( 'ltr', 'text direction' );
+	};
+
+	setLocaleData( initialData, initialDomain );
```

Comment:
Should we always be calling this, even if `initialData` and/or `initialDomain` are possibly `undefined` (per the documented types)?

Reply:
Related to this, are there specific use-cases in mind for setting the initial data? It seems like something which _could_ be useful, but just wondering if we might be [getting ahead of ourselves](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).

Reply:
I requested this. It seems helpful to be able to populate with initial data, but there's not much lost if we require package consumers to first create then set the data. I don't have strong feelings.

Reply:
It's still odd to me that we call `setLocaleData` if there is nothing to set. It _works_, but only by a combination of the fact that `data` is an optional argument of `setLocaleData` (pretty counter-intuitive), and that object spread is tolerant of undefined (i.e. `{ ...undefined }`), when instead we could just add an `if` here, and make `setLocaleData`'s `data` argument non-optional.

```suggestion
	if ( initialData ) {
		setLocaleData( initialData, initialDomain );
	}
```

Reply:
@jameslnewell Do you mind addressing this?

Reply:
I addressed this in #21182. I believe that's the remaining feedback that needed to be addressed.